pennylane/ops/op_math/prod.py

```diff
@@ -332,40 +336,48 @@ def adjoint(self):
     def arithmetic_depth(self) -> int:
         return 1 + max(factor.arithmetic_depth for factor in self.factors)
 
-    @classmethod
-    def _simplify_factors(cls, factors: Tuple[Operator]) -> Tuple[Operator]:
-        """Reduces the depth of nested factors.
+    def _simplify_factors(self, factors: Tuple[Operator]) -> Tuple[complex, Operator]:
+        """Reduces the depth of nested factors and groups identical factors.
 
         Returns:
-            Tuple[List[~.operation.Operator], List[~.operation.Operator]: reduced sum and non-sum
-            factors
+            Tuple[complex, List[~.operation.Operator]: tuple containing the global phase and a list
+            of the simplified factors
         """
-        new_factors = ()
+        new_factors = ProductFactorsGrouping()
 
         for factor in factors:
-            if isinstance(factor, Prod):
-                tmp_factors = cls._simplify_factors(factors=factor.factors)
-                new_factors += tmp_factors
-                continue
             simplified_factor = factor.simplify()
-            if isinstance(simplified_factor, Prod):
-                new_factors += tuple((factor,) for factor in simplified_factor.factors)
-            elif isinstance(simplified_factor, Sum):
-                new_factors += (simplified_factor.summands,)
-            elif not isinstance(simplified_factor, qml.Identity):
-                new_factors += ((simplified_factor,),)
-
-        return new_factors
+            new_factors.add(factor=simplified_factor)
+        new_factors.remove_factors(wires=self.wires)
```

Comment:
Remove ALL factors from the `new_factors._pauli_factors` and `new_factors._non_pauli_factors` dictionaries and add them to the `new_factors.factors` tuple.